### ISSUE TITLE
"Getting SHSH --> Saving SHSH failed" iPad 2 16GB WiFi (iOS 9.3.5 --> iOS 6.1.3)

### ISSUE THREAD

**Issue by Jhuang114** [MatthewPierson/Vieux#166]
Hello! I am attempting to downgrade an iPad 2 (2,1) from iOS 9.3.5 to iOS 6.1.3 through Vieux, but keep getting an error that "Saving SHSH Failed". I have tried on multiple devices but the error persists. I have already checked each time and the device's jailbreak IS activated and OpenSSH IS installed. Everything has been done accordingly, but the issue remains! Please help!

Computers: MacBook Pro 13" 2019 (macOS Catalina) and iMac 21.5" 2015 (macOS Mojave)
Downgrading Device: iPad 2 16GB WiFi (iOS 9.3.5 --> iOS 6.1.3)
Primary Error: "Getting SHSH --> Saving SHSH failed"
Jailbreak Status: Jailbroken w/ OpenSSH Installed

Transcript of Error:
Vieux - A tool for 32/64 Bit OTA downgrades
Current version is: 1.0.1
If you are using a 64 Bit device then connect it in DFU Mode
If you are using a 32 Bit device then just have it connected in NORMAL mode
Files cleaned.
/Users/Jonathan/Desktop/iPad2,1_6.1.3_10B329_Restore.ipsw is a zip archive!
Starting IPSW unzipping
Continuing...
IPSW found at given path...
Cleaning up old files...
Files cleaned.
Unzipping..
Starting iBSS/iBEC patching
Looks like you are downgrading an iPad2 to 6.1.3!
Please enter your devices IP address (Find it in WiFi settings):
*HIDDEN FOR PRIVACY*
Please enter the root password to your device (Default is 'alpine'):
Connecting to device via SSH...
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/paramiko/client.py:837: UserWarning: Unknown ssh-rsa host key for *HIDDEN FOR PRIVACY*: b'53ee9f6386025dde461102f521ada893'
  key.get_name(), hostname, hexlify(key.get_fingerprint())
Device is not running iOS 10.x, using normal TFP0 kloader...
Please press the home button on your device or unplug and replug it back in.
Waiting 10 seconds for you to do this.
Getting SHSH...
Saving SHSH failed.
Please try again and report the error + full logs if it persists.
Exiting...

**Comment by MatthewPierson** [comment]
Are you pressing the home button then unplugging the device and replugging it back in when prompted? The SHSH saving error usually happens when the user doesn't do those things so when Vieux tries to get the devices information, it can't.

**Comment by Jhuang114** [comment]
> Are you pressing the home button then unplugging the device and replugging it back in when prompted? The SHSH saving error usually happens when the user doesn't do those things so when Vieux tries to get the devices information, it can't.

Yes, I have done that each time I try it. As soon as the propt to press the home button comes up, I immediately press it. I've tried just pressing the button, I've tried just unplugging and plugging back in. Either way, the same error happens!

**Comment by Jhuang114** [comment]
> Are you pressing the home button then unplugging the device and replugging it back in when prompted? The SHSH saving error usually happens when the user doesn't do those things so when Vieux tries to get the devices information, it can't.

Another issue that I am also getting is the following. I am following everything perfectly up to the getting SHSH. Either it says that it was unable to get SHSH or that no device was found in DFU mode... even though the instructions say "If you are using a 32 Bit device then just have it connected in NORMAL mode"

Other Issue:
Getting SHSH...
ERROR: No Apple device in DFU Mode 0x1227 detected after 30.00 second timeout. Exiting.
Place the device into DFU mode again and re-run Vieux

**Comment by Jhuang114** [comment]
Yes, I have done that each time I try it. As soon as the propt to press the home button comes up, I immediately press it. I've tried just pressing the button, I've tried just unplugging and plugging back in. Either way, the same error happens!

**Comment by TheSatan1st** [comment]
Same here on iPad 2,1!

**Comment by napalkovn** [comment]
i have same problem, please help

**Comment by dnsmob** [comment]
i'm on ipad2,2 and i also get the DFU timeout error. i ran Catalina.sh and changed codesign on the binaries, all fine. i have the impression that vieux does save the SHSH and extracts whatever plists as required, it only fails to enter DFU. if there was a manual way of doing that, or using fururestore manually, that might be a solution.

**Comment by estorado** [comment]
Same here. The Ipad 2,2 just turns off immidatly after unplug it or press the home button and dont come up any more. Is it possible to set the iPad in dfu mode and continue with the process? It seems that the ssh connection get lost after disconnect the usb connection or pressing the home button

**Comment by UniDay** [comment]
Ok. I finaly did this quest )) I do not know if it really matter but! 
I did it under macOS High Sierra. iPad 2,1 (wifi) 9.3.5->6.1.3
first run every (3) script in Catalina.sh
then like in instruction ./vieux -i /Users/XXXXXXXXX/Downloads/iPad2\,1_6.1.3_10B329_Restore.ipsw
then when it prompt to unplug/plug device or press home button I disconnect cable from iPad then connect cable and then press home button and all works fine.

Messages was like below.

Please press the home button on your device or unplug and replug it back in.
Waiting 10 seconds for you to do this.
Getting SHSH...
Restoring...
Note that errors about 'BbSkeyId', 'FDR Client', 'BasebandFirmware Node' and 'ERROR: zip_name_locate: Firmware/all_flash/manifest' are not important.
Just ignore them and only report errors that actually stop the restore.
ERROR: Unable to connect to FDR client (-2)
ERROR: Failed to start FDR Ctrl channel
Files cleaned.
Finished! Enjoy your downgraded device :)
